tests/Validator/SingleFieldSubscriptionsTest.php

```diff
@@ -0,0 +1,190 @@
+          barks
+        }
+      }
+        '
+        );
+    }
+
+    /**
+     * @see it('valid single field anonymous subscription')
+     */
+    public function testValidSingleFieldAnonymousSubscription() : void
+    {
+        $this->expectPassesRule(
+            new SingleFieldSubscription(),
+            '
+      subscription {
+        catSubscribe {
+          meows
+        }
+      }
+        '
+        );
+    }
+
+    /**
+     * @see it('valid single field subscription')
+     */
+    public function testValidSingleFieldSubscriptionWithMultipleResultFields() : void
+    {
+        $this->expectPassesRule(
+            new SingleFieldSubscription(),
+            '
+      subscription {
+        catSubscribe {
+          meows
+          meowVolume
+        }
+      }
+        '
+        );
+    }
+
+    /**
+     * @see it('invalid multiple field subscription')
+     */
+    public function testInvalidMultipleFieldSubscription() : void
+    {
+        $this->expectFailsRule(
+            new SingleFieldSubscription(),
+            '
+      subscription sub {
+        catSubscribe {
+          meows
+        }
+        barkSubscribe {
+          barks
+        }
+      }
+        ',
+            [$this->multipleFieldsInOperation('sub', [6, 9])]
+        );
+    }
+
+    /**
+     * @see it('invalid multiple field anonymous subscription')
+     */
+    public function testInvalidMultipleFieldAnonymousSubscription() : void
+    {
+        $this->expectFailsRule(
+            new SingleFieldSubscription(),
+            '
+      subscription {
+        catSubscribe {
+          meows
+        }
+        barkSubscribe {
+          barks
+        }
+      }
+        ',
+            [$this->multipleFieldsInOperation(null, [6, 9])]
+        );
+    }
+
+    /**
+     * @see it('invalid many fields subscription')
+     */
+    public function testInvalidManyFieldsSubscription() : void
+    {
+        $this->expectFailsRule(
+            new SingleFieldSubscription(),
+            '
+      subscription sub {
+        first: catSubscribe {
+          meows
+        }
+        second: catSubscribe {
+          meows
+        }
+        third: catSubscribe {
+          meows
+        }
+      }
+        ',
+            [$this->multipleFieldsInOperation('sub', [6, 9], [9, 9])]
+        );
+    }
+
+    /**
+     * @see it('invalid many fields anonymous subscription')
+     */
+    public function testInvalidManyFieldAnonymousSubscription() : void
+    {
+        $this->expectFailsRule(
+            new SingleFieldSubscription(),
+            '
+      subscription {
+        first: catSubscribe {
+          meows
+        }
+        second: catSubscribe {
+          meows
+        }
+        third: catSubscribe {
+          meows
+        }
+      }
+        ',
+            [$this->multipleFieldsInOperation(null, [6, 9], [9, 9])]
+        );
+    }
+
+    private function multipleFieldsInOperation($operationName, ...$locations)
```

Comment:
```suggestion
    /**
     * @param array<int, int> ...$locations A tuple of line and column
     */
    private function multipleFieldsInOperation(?string $operationName, array ...$locations): string
```

Not too sure if PHPStan will accept that.